lib/seahorn/BvOpSem2.cc

```diff
@@ -555,6 +560,42 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
     setValue(*CS.getInstruction(), res);
   }
 
+  // TODO: visit.assert.if.not is not supported
+  void visitAssertIf(CallSite CS) {
+    auto I = CS.getInstruction();
+    Expr ante = lookup(*CS.getArgument(0));
+    Expr conseq = lookup(*CS.getArgument(1));
+    tribool anteRes = solveWithConstraints(ante);
+    // if ante is unsat then report false and bail out
+    if (!anteRes) {
+      LOG("opsem", ERR << "Antecedent is unsat/undet: " << *I << "\n");
+      return; // return early
+    }
+    Expr e = boolop::land(ante, boolop::lneg(conseq));
+    tribool conseqRes = solveWithConstraints(e);
+    if (conseqRes) {
+      LOG("opsem", ERR << "Consequent is sat/undet: " << *I << "\n");
```

Comment:
need options for vacuity and for incremental checking. Will need to time how long they take. We might need vacuity + final check, but not vacuity + incremental + final

Reply:
done